### ISSUE TITLE
Misleading error message when calling URL that wasn't mocked 

### ISSUE THREAD

**Issue by jumarko** [myfreeweb/clj-http-fake#36]
If I evaluate following code
```
(with-fake-routes
  {"http://google.com"
   (fn [request]
     {:status 200 :headers {} :body "R1BWm0"})}
  (:body (http/get "http://google.cz/")))
```
I get following error:
```
1. Unhandled clojure.lang.ArityException
   Wrong number of args (4) passed to: fake/try-intercept

                  AFn.java:  429  clojure.lang.AFn/throwArity
                  AFn.java:   44  clojure.lang.AFn/invoke
                  Var.java:  394  clojure.lang.Var/invoke
                  AFn.java:  165  clojure.lang.AFn/applyToHelper
                  Var.java:  700  clojure.lang.Var/applyTo
                  core.clj:  648  clojure.core/apply
                  core.clj:  641  clojure.core/apply
                 hooke.clj:   40  robert.hooke/compose-hooks/fn
               RestFn.java:  137  clojure.lang.RestFn/applyTo
                  core.clj:  646  clojure.core/apply
                  core.clj:  641  clojure.core/apply
                 hooke.clj:   46  robert.hooke/run-hooks
                 hooke.clj:   45  robert.hooke/run-hooks
                 hooke.clj:   54  robert.hooke/prepare-for-hooks/fn/fn
               RestFn.java:  137  clojure.lang.RestFn/applyTo
            AFunction.java:   29  clojure.lang.AFunction$1/doInvoke
               RestFn.java:  436  clojure.lang.RestFn/invoke
                  core.clj:  322  clj-http.core/request
                  core.clj:  321  clj-http.core/request
                  fake.clj:  173  clj-http.fake/try-intercept
                  fake.clj:  159  clj-http.fake/try-intercept
                  Var.java:  383  clojure.lang.Var/invoke
                  AFn.java:  156  clojure.lang.AFn/applyToHelper
                  Var.java:  700  clojure.lang.Var/applyTo
                  core.clj:  648  clojure.core/apply
                  core.clj:  641  clojure.core/apply
                 hooke.clj:   40  robert.hooke/compose-hooks/fn
               RestFn.java:  137  clojure.lang.RestFn/applyTo
                  core.clj:  646  clojure.core/apply
                  core.clj:  641  clojure.core/apply
                 hooke.clj:   46  robert.hooke/run-hooks
                 hooke.clj:   45  robert.hooke/run-hooks
                 hooke.clj:   54  robert.hooke/prepare-for-hooks/fn/fn
               RestFn.java:  137  clojure.lang.RestFn/applyTo
            AFunction.java:   29  clojure.lang.AFunction$1/doInvoke
               RestFn.java:  408  clojure.lang.RestFn/invoke
                  Var.java:  379  clojure.lang.Var/invoke
                client.clj: 1013  clj-http.client/wrap-request-timing/fn
                client.clj: 1046  clj-http.client/wrap-async-pooling/fn
               headers.clj:  147  clj-http.headers/wrap-header-map/fn
                client.clj:  773  clj-http.client/wrap-query-params/fn
                client.clj:  796  clj-http.client/wrap-basic-auth/fn
                client.clj:  813  clj-http.client/wrap-oauth/fn
                client.clj:  833  clj-http.client/wrap-user-info/fn
                client.clj:  965  clj-http.client/wrap-url/fn
                client.clj:  344  clj-http.client/wrap-redirects/fn
                client.clj:  397  clj-http.client/wrap-decompression/fn
                client.clj:  593  clj-http.client/wrap-input-coercion/fn
                client.clj:  648  clj-http.client/wrap-additional-header-parsing/fn
                client.clj:  537  clj-http.client/wrap-output-coercion/fn
                client.clj:  232  clj-http.client/wrap-exceptions/fn
                client.clj:  691  clj-http.client/wrap-accept/fn
                client.clj:  713  clj-http.client/wrap-accept-encoding/fn
                client.clj:  674  clj-http.client/wrap-content-type/fn
                client.clj:  915  clj-http.client/wrap-form-params/fn
                client.clj:  950  clj-http.client/wrap-nested-params/fn
                client.clj:  849  clj-http.client/wrap-method/fn
               cookies.clj:  131  clj-http.cookies/wrap-cookies/fn
                 links.clj:   63  clj-http.links/wrap-links/fn
                client.clj:  976  clj-http.client/wrap-unknown-host/fn
                client.clj: 1149  clj-http.client/request*
                client.clj: 1142  clj-http.client/request*
                client.clj: 1155  clj-http.client/get
                client.clj: 1151  clj-http.client/get
               RestFn.java:  410  clojure.lang.RestFn/invoke
...
             Compiler.java: 6927  clojure.lang.Compiler/eval
             Compiler.java: 6890  clojure.lang.Compiler/eval
                  core.clj: 3105  clojure.core/eval
                  core.clj: 3101  clojure.core/eval
                  main.clj:  240  clojure.main/repl/read-eval-print/fn
                  main.clj:  240  clojure.main/repl/read-eval-print
                  main.clj:  258  clojure.main/repl/fn
                  main.clj:  258  clojure.main/repl
                  main.clj:  174  clojure.main/repl
               RestFn.java: 1523  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   87  clojure.tools.nrepl.middleware.interruptible-eval/evaluate/fn
                  AFn.java:  152  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  core.clj:  646  clojure.core/apply
                  core.clj: 1881  clojure.core/with-bindings*
                  core.clj: 1881  clojure.core/with-bindings*
               RestFn.java:  425  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   85  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:   55  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:  222  clojure.tools.nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
    interruptible_eval.clj:  190  clojure.tools.nrepl.middleware.interruptible-eval/run-next/fn
                  AFn.java:   22  clojure.lang.AFn/run
   ThreadPoolExecutor.java: 1142  java.util.concurrent.ThreadPoolExecutor/runWorker
   ThreadPoolExecutor.java:  617  java.util.concurrent.ThreadPoolExecutor$Worker/run
               Thread.java:  745  java.lang.Thread/run

```
This error message is quite confusing. It would be great to know what's happening in this case.

**Comment by jumarko** [comment]
I've noticed that the latest version from master is much better than 1.2.
When do you plan to release 1.3?

**Comment by myfreeweb** [comment]
I wasn't planning anything, I haven't actively worked on anything Clojure related in a long time :D

Released 1.0.3 (not 1.3!). Is everything okay in the new version?

**Comment by jumarko** [comment]
Of cource, it's 1.0.3, not 1.3. :).
And works great. Thanks!
